nimble/host/src/ble_l2cap_coc.c

```diff
@@ -255,6 +255,11 @@ ble_l2cap_coc_rx_fn(struct ble_l2cap_chan *chan)
         return 0;
     }
 
+    if (OS_MBUF_PKTLEN(rx->sdu) > rx->data_offset) {
```

Comment:
This seems to be too late.
I think we should add checks somewhere around line 201 and in case of fragmentation we should add additional checks  around 229.

In addition, when we decide to disconnect, we probably need to clear already received data.